nautobot_circuit_maintenance/handle_notifications/sources.py

```diff
@@ -309,7 +309,11 @@ def open_session(self):
         """Open session to IMAP server."""
         if not self.session:
             self.session = imaplib.IMAP4_SSL(self.imap_server, self.imap_port)
-            self.session.login(self.account, self.password)
+            try:
+                self.session.login(self.account, self.password)
+            except imaplib.IMAP4.error:
+                self.session = None
```

Comment:
Should `self.close_session` also set `self.session = None` after closing the session, or is the asymmetry intentional?